src/riak_pb_ts_codec.erl

```diff
@@ -180,7 +182,8 @@ decode_cells([#tscell{binary_value    = undefined,
                       timestamp_value = undefined,
                       boolean_value   = undefined,
                       set_value       = Set,
-                      map_value       = undefined} | T], Acc) ->
+                      map_value       = undefined} | T], Acc)
+  when is_list(Set) ->
```

Comment:
I'm checking `length(Set) > 0` in 5dc3e8f now; that way the last clause is very much undefined and not the `set_value` clause being undefined by chance.